hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java

```diff
@@ -8604,6 +8605,13 @@ IOException throwOnInterrupt(Throwable t) {
   @Override
   public void onConfigurationChange(Configuration conf) {
     this.storeHotnessProtector.update(conf);
+    // update coprocessorHost if the configuration has changed.
+    if (CoprocessorConfigurationUtil.checkConfigurationChange(getReadOnlyConfiguration(), conf,
+      CoprocessorHost.REGION_COPROCESSOR_CONF_KEY,
+      CoprocessorHost.USER_REGION_COPROCESSOR_CONF_KEY)) {
+      LOG.info("Update the system coprocessors because the configuration has changed");
+      this.coprocessorHost = new RegionCoprocessorHost(this, rsServices, conf);
```

Comment:
HRegion.decorateRegionConfiguration(conf) ?

Reply:
ack, thanks for spotting it.